### ISSUE TITLE
False positives for first-nested preceded by a shared-line comment in *-empty-line-before

### ISSUE THREAD

**Issue by rudywaltz** [stylelint/stylelint#2764]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where i combine comment and sass extend.

> Which rule, if any, is this issue related to?

e.g. `at-rule-empty-line-before`

> What CSS is needed to reproduce this issue?

e.g.

```scss
&:checked {
  + .e-accordion__title:nth-child(n) { //nth-child: safari not repaint bug fix
    @extend .e-accordion__title-active;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "at-rule-empty-line-before": [ "always", {
      "except": [ "blockless-after-blockless", "first-nested" ],
      "ignore": ["after-comment"],
      "ignoreAtRules": ["else"]
    }],
  }
}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?
CLI with `stylelint app/components/**/**/*.scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No, it Can be reproduced with normal css comment.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
app/components/components/accordion/_accordion.scss
 99:11  ✖  Expected empty line before at-rule   at-rule-empty-line-before
```

**Comment by jeddy3** [comment]
@rudywaltz Thanks for the report and for using the template.

This is reproducible with:

```css
a { /* comment */
  @at-rule;
}
```

And:

```json
{
  "rules": {
    "at-rule-empty-line-before": [ "always", {
      "except": ["first-nested"],
    }],
  }
}
```

The at-rule, rather than the _shared-line_ comment, should be considered the "first-nested" node.

Labelling as `help wanted` and `bug`. @rudywaltz Feel free to [contribute a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by jeddy3** [comment]
This is [seemingly applicable](https://github.com/stylelint/stylelint/issues/2824) to all `*-empty-line-before` rules.
